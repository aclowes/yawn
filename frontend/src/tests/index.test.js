import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import ReactTestRenderer from 'react-test-renderer';

jest.mock('react-dom', () => ({render: jest.fn()}));

import {YawnRouter, NotFound} from '../index';
import {mockAPI} from './mocks'

it('router renders', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<YawnRouter />);
  const result = renderer.getRenderOutput();
  // this doesn't work for some reason...
  // expect(result.type).toBe('Router');
});

it('404 renders', () => {
  mockAPI();
  const component = ReactTestRenderer.create(<NotFound />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('ReactDOM is called', () => {
  expect(ReactDOM.render).toHaveBeenCalled();
});
