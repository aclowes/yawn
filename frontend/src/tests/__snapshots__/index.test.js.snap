// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 renders 1`] = `
<h3>
  Not Found (404)
</h3>
`;
